### ISSUE TITLE
Substitute version/branch names in schema

### ISSUE THREAD

**Issue by duncandewhurst** [Open-Telecoms-Data/open-fibre-data-standard#153]
Currently, links in the schema are hardcoded to the `latest` branch. OCDS has an approach to substituting version/branch names.

**Comment by ghost** [comment]
#156 is related. 

I haven't looked at OCDS approach. 

In BODS we took an approach of trying to move any such links out of description fields. Other tools may use the JSON Schema, and we can't be clear how that will work and whether it will break things there. (In particular, in BODS we previously had relative links but there were 2 places they became visibile in the standard and one worked and one didn't.)

Instead we tell the widgets drawing the docs to add links themselves eg see https://github.com/openownership/data-standard/blob/0.2.0/docs/schema/reference.rst#address and the :externallinks: key

In the docs this appears as a seperate link eg https://standard.openownership.org/en/0.2.0/schema/reference.html#address

**Comment by duncandewhurst** [comment]
OCDS's approach is to replace the language and version placeholders in the schema files in the GitHub repository as part of the build/deploy process ([docs](https://ocds-standard-development-handbook.readthedocs.io/en/latest/standard/schema.html#the-schema)) and to push the canonical processed version to a folders on OCP's server, e.g. https://standard.open-contracting.org/schema/1__1__5/. Those are the schema files that should be used by other tools as described on https://standard.open-contracting.org/latest/en/schema/release/#release-schema. I think that is preferable because users get a direct link to the referenced codelist, for example, rather than needing to browse/search the codelist reference, and because those links are available to other tools/users.

For OFDS, we don't have a separate server, but maybe we can still do something as part of the docs build process on readthedocs, i.e. making substitutions and pushing the canonical versions to a sub-directory of the built docs folder.

**Comment by ghost** [comment]
> For OFDS, we don't have a separate server, but maybe we can still do something as part of the docs build process on readthedocs

Yes, we could. This is getting into a wider conversation of how we build standards, and I'm not definitely saying we should do it the BODS way, just thought it would be interesting to consider all the options we have.

**Comment by duncandewhurst** [comment]
Noting that I've got this working on Read the Docs in https://github.com/GFDRR/rdl-standard/pull/100

**Comment by duncandewhurst** [comment]
The proposal is to reuse the approach from RDLS to substitute version/branch names in schema so that links in the schema work on development branches.

**Comment by duncandewhurst** [comment]
https://github.com/Open-Telecoms-Data/open-fibre-data-standard/pull/325 implements {{version}} placeholder replacement in `network-schema.json`.

Still to do:

- [x] replace placeholders in `network-schema.gpkg`
- [x] replace placeholders in `network-schema.csv`

Currently, those files are produced by the pre-commit script in manage.py, whereas the placeholder substitution is implemented in `conf.py` and happens as part of the docs build.

- [x] update the linkcheck action to remove false positives

Currently, linkcheck reports lots of broken links, because it runs it's own build of the docs, which doesn't trigger the placeholder replacement.
